pkgs/tools/misc/beam-book/default.nix

```
@@ -0,0 +1,98 @@
+{ lib, stdenv
+, fetchFromGitHub
+, wget, asciidoctor, ditaa, graphviz
+, gitMinimal
+, erlang
+, jdk
+, rsync
+, rouge
+}:
+
+let
+  pname = "the-beam-book";
+  version = "0.0.14.fix";
+
+  bookDirName = "${pname}-${version}";
+
+  book = fetchFromGitHub {
+    name = bookDirName;
+    owner = "happi";
+    repo = "theBeamBook";
+    rev = version;
+    sha256 = "06r19063a9s6hg84garn83rn4cnnkvs58x3j68zvf94mwsmfs86h";
+    leaveDotGit = true;
+  };
+
+  erlangGenOp = fetchFromGitHub {
+    name = "erlang.tar.gz";
+    owner = "erlang";
+    repo = "otp";
+    rev = "master";
+    sha256 = "1qfdry8wdba3abfsa1qkkcglww186j8dyl40si1vrwsc7f221r1d";
+  };
+
+in stdenv.mkDerivation rec {
+  inherit pname version;
+
+  srcs = [ book erlangGenOp ];
+
+  outputs = [ "out" "pdf" ];
+
+  sourceRoot = bookDirName;
+
+  postUnpack = ''
+    cp ${erlangGenOp}/lib/compiler/src/genop.tab ${bookDirName}/
+  '';
+
+  patches = [
+    # Cannot call wget in build environment. File is fetched during unpackPhase
+    ./remove-wget.patch
+
+    # bin/gitlog.sh script calls `git shortlog` without commit parameter, which
+    # is problematic in non-interactive shell sessions.
+    # https://stackoverflow.com/a/43042363
+    ./add-HEAD-to-git-shortlog.patch
+  ];
+
+  # Patch shebangs after we apply all patches
+  postPatch = ''
+    patchShebangs .
+  '';
+
+  dontConfigure = true;
+
+  preBuild = ''
+    mkdir -p $sourceRoot/chapters
+  '';
+
+  nativeBuildInputs = [
+    wget
+    asciidoctor
+    ditaa
+    graphviz
+    rouge
+    erlang
+    jdk
+    gitMinimal
+    rsync
+  ];
+
+  doCheck = false;
+
+  installPhase = ''
+    mkdir -p $out
+    mkdir -p $pdf
+    cp -r site/* $out/
+    cp -r beam-book.pdf $pdf/
+    ln -s -T "$pdf/beam-book.pdf" "$out/beam-book.pdf"
```

```suggestion
    cp -r beam-book.pdf $out/share/beam-book/beam-book.pdf
```
we do not want to pollute / when installing this globally.

---

3 things:
1. I wanted to separate website (HTML) and PDF output so that they would not confuse each other.
2. Installing the PDF to `$out` would amount to removing the `$pdf` output.
3. I did not make that deep directory structure, because no binaries are output. Just a website (HTML) and a PDF.

---

> I did not make that deep directory structure, because no binaries are output. Just a website (HTML) and a PDF.

but we need this. Otherwise / is polluted with this file on NixOS if the package is installed system wide.